java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/RequestContextAdapter.java

```diff
@@ -30,7 +30,7 @@
  */
 public class RequestContextAdapter implements RequestContext {
   public static final Context.Key<RequestContext> REQUEST_CONTEXT_KEY =
-          Context.key("arrow-flight-request-context");
+          Context.keyWithDefault("arrow-flight-request-context", new RequestContextAdapter());
```

Comment:
Sorry - now that I think about this again, this is shared mutable state, so wouldn't it be possible (though unlikely) for different calls to persist data into this instance of RequestContextAdapter? It might be better to leave this nullable and check usages of it instead.

Reply:
Done